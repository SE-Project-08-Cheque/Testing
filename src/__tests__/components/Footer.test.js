// import dependencies
import React from 'react';
// import react dom
import ReactDOM from 'react-dom';
// import react-testing methods
import {render, cleanup, fireEvent} from '@testing-library/react';
// add custom jest matchers from jest-dom
import "@testing-library/jest-dom/extend-expect";
// import renderer for take snapshots
import renderer from 'react-test-renderer';

// import the component for testing
import Footer from '../../components/Footer';
import { MemoryRouter } from 'react-router-dom';

afterEach(cleanup);

it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<MemoryRouter><Footer /></MemoryRouter>, div)
})

it("renders footer correctly", () => {
    const {getByTestId} = render(<MemoryRouter><Footer /></MemoryRouter>)
    expect(getByTestId('subcription-heading')).toBeTruthy();
    expect(getByTestId('subcription-heading')).toHaveTextContent("Do you need technical guidance");
    expect(getByTestId('subcription-text')).toBeTruthy();
    expect(getByTestId('subcription-text')).toHaveTextContent("Send us what you want to know");
    expect(getByTestId('send')).toBeTruthy();
    expect(getByTestId('send')).toHaveTextContent("Send us");


})


it("matches snapshot", () => {
    const tree = renderer.create(<MemoryRouter><Footer /></MemoryRouter>).toJSON();
    expect(tree).toMatchSnapshot();
})