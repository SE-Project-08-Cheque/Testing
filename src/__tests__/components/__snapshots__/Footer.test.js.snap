// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="footer-container"
>
  <section
    className="footer-subscription"
  >
    <p
      className="footer-subscription-heading"
      data-testid="subcription-heading"
    >
      Do you need technical guidance?
    </p>
    <p
      className="footer-subscription-text"
      data-testid="subcription-text"
    >
      Send us what you want to know. We will contact you as soon as possible.
    </p>
    <div
      className="input-areas"
    >
      <form>
        <button
          className="chakra-button css-oqfhsh"
          data-testid="send"
          type="button"
        >
          Send us
          <span
            className="chakra-button__icon css-1oeb2oe"
          >
            <svg
              aria-hidden={true}
              className="chakra-icon css-onkibi"
              focusable={false}
              viewBox="0 0 24 24"
            >
              <path
                d="M12 4l-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
                fill="currentColor"
              />
            </svg>
          </span>
        </button>
      </form>
    </div>
  </section>
  <div
    class="footer-links"
  >
    <div
      className="footer-link-wrapper"
    >
      <div
        class="footer-link-items"
      >
        <h2>
          About Us
        </h2>
        <a
          href="/"
          onClick={[Function]}
        >
          Secured Service
        </a>
        <a
          href="/"
          onClick={[Function]}
        >
          24 / 7 Transactions
        </a>
        <a
          href="/"
          onClick={[Function]}
        >
          Sustainability
        </a>
        <a
          href="/"
          onClick={[Function]}
        >
          Terms of Service
        </a>
      </div>
      <div
        class="footer-link-items"
      >
        <h2>
          Contact Us
        </h2>
        <a
          href="/"
          onClick={[Function]}
        >
          Contact
        </a>
        <a
          href="/"
          onClick={[Function]}
        >
          Support
        </a>
        <a
          href="/"
          onClick={[Function]}
        >
          Customer Care
        </a>
        <a
          href="/"
          onClick={[Function]}
        >
          10000+ Users
        </a>
      </div>
    </div>
    <div
      className="footer-link-wrapper"
    >
      <div
        class="footer-link-items"
      >
        <h2>
          Social Media
        </h2>
        <a
          href="/"
          onClick={[Function]}
        >
          Instagram
        </a>
        <a
          href="/"
          onClick={[Function]}
        >
          Facebook
        </a>
        <a
          href="/"
          onClick={[Function]}
        >
          Youtube
        </a>
        <a
          href="/"
          onClick={[Function]}
        >
          Twitter
        </a>
      </div>
      <div
        class="footer-link-items"
      >
        <p
          className="chakra-text css-kbn7if"
        >
          Bank of Ceylon
        </p>
        <p
          className="chakra-text css-zjak30"
        >
          No 1, BOC Square, 
        </p>
        <p
          className="chakra-text css-kbn7if"
        >
          Ceylon Mawatha,
        </p>
        <p
          className="chakra-text css-kbn7if"
        >
          Colombo 01,Sri Lanka.
        </p>
        <p
          className="chakra-text css-kbn7if"
        >
          Tel : +94112446790
        </p>
        <p
          className="chakra-text css-kbn7if"
        >
          Email : cheque@boc.lk
        </p>
        <p
          className="chakra-text css-kbn7if"
        />
      </div>
    </div>
  </div>
  <section
    class="social-media"
  >
    <div
      class="social-media-wrap"
    >
      <div
        class="footer-logo"
      >
        <a
          className="social-logo"
          href="/"
          onClick={[Function]}
        >
          Online Cheque
          <i
            class="fab fa-typo3"
          />
        </a>
      </div>
      <small
        class="website-rights"
      >
        Â© 2021, All Rights Reserved | Designed by University of Moratuwa
      </small>
      <div
        class="social-icons"
      >
        <a
          aria-label="Facebook"
          class="social-icon-link facebook"
          href="/"
          onClick={[Function]}
          target="_blank"
        >
          <i
            class="fab fa-facebook-f"
          />
        </a>
        <a
          aria-label="Instagram"
          class="social-icon-link instagram"
          href="/"
          onClick={[Function]}
          target="_blank"
        >
          <i
            class="fab fa-instagram"
          />
        </a>
        <a
          aria-label="Youtube"
          class="social-icon-link youtube"
          href="/"
          onClick={[Function]}
          target="_blank"
        >
          <i
            class="fab fa-youtube"
          />
        </a>
        <a
          aria-label="Twitter"
          class="social-icon-link twitter"
          href="/"
          onClick={[Function]}
          target="_blank"
        >
          <i
            class="fab fa-twitter"
          />
        </a>
        <a
          aria-label="LinkedIn"
          class="social-icon-link twitter"
          href="/"
          onClick={[Function]}
          target="_blank"
        >
          <i
            class="fab fa-linkedin"
          />
        </a>
      </div>
    </div>
  </section>
</div>
`;
